android/app/src/main/java/com/microsoft/codepush/common/CodePushStatusReportIdentifier.java

```diff
@@ -0,0 +1,116 @@
+ * If both are exist it means that update report was saved, if only version label exists it means that binary report was saved.
+ * There shouldn't be a situation when version label doesn't exist.
+ * <p>
+ * Identifier can be serialized into string for saving on device and deserialized from it.
+ */
+public class CodePushStatusReportIdentifier {
+
+    /**
+     * Deployment key.
+     */
+    private String deploymentKey;
+
+    /**
+     * Version label.
+     */
+    private String versionLabel;
+
+    /**
+     * Constructor for identifier of binary report.
+     *
+     * @param versionLabel version label.
+     */
+    public CodePushStatusReportIdentifier(@NonNull String versionLabel) {
+        this.versionLabel = versionLabel;
+    }
+
+    /**
+     * Constructor for identifier of update report.
+     *
+     * @param deploymentKey deployment key.
+     * @param versionLabel  version label.
+     */
+    public CodePushStatusReportIdentifier(@NonNull String deploymentKey, @NonNull String versionLabel) {
+        this.deploymentKey = deploymentKey;
+        this.versionLabel = versionLabel;
+    }
+
+    /**
+     * Deserializes identifier from string.
+     *
+     * @param stringIdentifier input string.
+     * @return {@link CodePushStatusReportIdentifier} instance if it could be parsed, <code>null</code> otherwise.
+     */
+    public static CodePushStatusReportIdentifier fromString(String stringIdentifier) {
+        String[] parsedIdentifier = stringIdentifier.split(":");
+        if (parsedIdentifier.length > 0) {
+            return new CodePushStatusReportIdentifier(parsedIdentifier[0]);
+        } else if (parsedIdentifier.length > 1) {
+            return new CodePushStatusReportIdentifier(parsedIdentifier[0], parsedIdentifier[1]);
+        } else {
+            return null;
+        }
+    }
+
+    /**
+     * Serializes identifier to string.
+     *
+     * @return serialized identifier.
+     */
+    @Override
+    public String toString() {
+        if (versionLabel != null) {
+            if (deploymentKey == null) {
+                return versionLabel;
+            } else {
+                return deploymentKey + ":" + versionLabel;
+            }
+        } else {
+            return null;
+        }
+    }
+
+    /**
+     * Indicates whether identifier has deployment key.
+     *
+     * @return <code>true</code> if identifier has deployment key, <code>false</code> otherwise.
+     */
+    public boolean hasDeploymentKey() {
+        return deploymentKey != null;
+    }
+
+    /**
+     * Gets deployment key.
+     *
+     * @return deployment key.
+     */
+    public String getDeploymentKey() {
+        return deploymentKey;
+    }
+
+    /**
+     * Gets version label.
+     *
+     * @return version label.
+     */
+    public String getVersionLabel() {
+        return versionLabel;
+    }
+
+    /**
+     * Gets version label or empty string (<code>""</code>) if it equals null.
```

Comment:
Do we really need this `(<code>""</code>) ` clarification?

Reply:
@annakocheshkova Sure, why not? This is actually a short code fragment.

Reply:
@ruslan-bikkinin I think just "empty line" is enough for understanding. No need to clutter the comment here